pkgs/applications/virtualization/lima/bin.nix

```diff
@@ -57,11 +57,18 @@ stdenvNoCC.mkDerivation {
     chmod +x $out/bin/limactl
     wrapProgram $out/bin/limactl \
       --prefix PATH : ${lib.makeBinPath [ qemu ]}
+    runHook postInstall
+  '';
+
+  fixupPhase = ''
```

Comment:
This is broken. `fixupPhase` does [quite a lot of stuff](https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/fixup-phase.md) and when they override it like that, it will no longer happen. If it really needs to go to `fixupPhase`, it should be put it in `postFixup` hook.

Reply:
@jtojnar This does not work unfortunately because the postFixup runs before the binary patching. Anyway I think I found a way to manually patch this.

Reply:
postFixup should not run before binary patching. In case this really doesn't work, try using distPhase or postDistPhase

Reply:
@SuperSandro2000 Wow, both `distPhase` and `postDist` work. I have never heard about this phase? What does it do and when is it run?

Reply:
@SuperSandro2000 `runHook` first runs implicit hook, then implicit ones, so `postFixup` will be called before autoPatchelfHook's explicit `autoPatchelfPostFixup` hook. I've added another explicit postFixupHook from installPhase. I hope that keeps abstractions somewhat solid. I don't like the idea of relying on `distPhase` not doing anything useful and running the last in the list.